docs/architecture/Overview.md

```diff
@@ -0,0 +1,164 @@
+The other end of the component layer is a set of highly customizable (mostly) pure UI components. These UI components are agnostic of ACS concepts (and in particular the stateful clients) and are entirely driven by data and callback props defined within this packlet. See [this related doc](./CustomizableComponent.md) for some inspiration on how to design good UI components.
+
+### Bindings
+
+The two parts of the component layer - stateful clients and UI components define their own API. The binding layer provides the bridge that connects these two.
+
+- [Selectors](./WritingSelectors.md) effeciently bind UI component data props to specific nodes in the immutable state exposed by the stateful layer. Selectors are setup to (1) avoid unnecessary component rerenders, and (2) avoid costly computation to determine when the component needs to rerender.
+- Handlers bind UI component callback props to methods exposed by the stateful layer.
+
+For both selectors and handlers, the library exposes a very simple API for customers wanting to connect the UI components and the stateful clients without any customization - a single `usePropsFor` hook. The library also exposes a more verbose API for sophisticated use-cases. This progressive disclosure of detail allows customers to do simple things easily without limiting more complex scenarios.
+
+## Composite layer
+
+The library includes three UI composites, that bring together the UI components into somewhat customizable hero scenarios:
+
+- [`CallComposite`](../../packages/react-composites/src/composites/CallComposite/) implements a single PSTN / 1:N / group / teams interop call flow; powered by `CallAdapter`.
+- [`ChatComposite`](../../packages/react-composites/src/composites/ChatComposite/) implements a single chat session flow; powered by `ChatAdapter`.
+- [`CallWithChatComposite`](../../packages/react-composites/src/composites/CallWithChatComposite/) brings together the two composites above to support a single call flow that includes an associated chat thread, powered by `CallWithChatAdapter`.
+
+An application that leverages one of the composites looks as follows:
+
+        ┌─────────────────────────────────────────────┐
+        │ Contoso Application                         │
+        │  * Create adapters                          │
+        │  * Create composite using the adapters      │
+        │   ┌─────────────────────────────────────┐   │
+        │   │ UI Composites                       │   │
+        │   │  * CallComposite                    │   │
+        │   │  * ChatComposite                    │   │
+        │   │  * CallWithChatComposite            │   │
+        │   │    ┌─────────────────────────────┐  │   │
+        │   │    │ UI Components               │  │   │
+        │   │    └─┬───────────────────────▲───┘  │   │
+        │   │      │                       │      │   │
+        │   │    ┌─┼───────────────────────┼───┐  │   │
+        │   │    │ │   Component Bindings  │   │  │   │
```

Comment:
Would be nice to mention somewhere that bindings/components are not exposed api, but just implementation detail 
(I can tell it from those borders, but not sure if all readers can)

Reply:
Yeah. I was wishing for colored boxes :D